pkg/liqoctl/install/provider/provider_suite_test.go

```diff
@@ -19,9 +19,45 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/spf13/pflag"
+
+	"github.com/liqotech/liqo/pkg/liqoctl/install/provider"
 )
 
 func TestProvider(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "Provider Suite")
 }
+
+func newFlagSet() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	flags.String("cluster-name", "", "")
+	flags.Bool("generate-name", false, "")
+	flags.String("reserved-subnets", "", "")
+	return flags
+}
+
+var _ = Describe("ValidateGenericCommandArguments", func() {
+	It("should accept a valid name", func() {
+		flags := newFlagSet()
+		clusterName := "test-cluster123"
+		Expect(flags.Set("cluster-name", clusterName)).To(Succeed())
+		provider := provider.GenericProvider{}
+		Expect(provider.ValidateGenericCommandArguments(flags)).To(Succeed())
+		Expect(provider.ClusterName).To(Equal(clusterName))
```

Comment:
You can use BeforeEach/JustBeforeEach blocks (or DescribeTables) to avoid code duplication.